src/content/app/genome-browser/state/browser-general/browserGeneralSelectors.ts

```diff
@@ -54,28 +54,27 @@ export const getBrowserTrackState = (
   state: RootState,
   params: {
     genomeId: string;
-    categoryName: string;
-    trackId: string;
   } & (
     | {
         objectId: string;
         tracksGroup: 'objectTracks';
       }
     | {
+        categoryName: string;
         tracksGroup: 'commonTracks';
+        trackId: string;
       }
   )
 ) => {
-  const { genomeId, tracksGroup, categoryName, trackId } = params;
+  const { genomeId, tracksGroup } = params;
   const allBrowserTrackStates = getBrowserTrackStates(state);
   const savedTrackStatus =
     tracksGroup === 'objectTracks'
-      ? allBrowserTrackStates?.[genomeId]?.[tracksGroup]?.[params.objectId]?.[
-          categoryName
-        ]?.[trackId]
-      : allBrowserTrackStates?.[genomeId]?.[tracksGroup]?.[categoryName]?.[
-          trackId
-        ];
+      ? allBrowserTrackStates?.[genomeId]?.[tracksGroup]?.[params.objectId]
+          ?.status
+      : allBrowserTrackStates?.[genomeId]?.[tracksGroup]?.[
+          params.categoryName
+        ]?.[params.trackId];
```

Comment:
Should we consider replacing this with `if else` to make it more readable?

Reply:
✅